src/Angor/Avalonia/AngorApp/Sections/Browse/ProjectLookup/ProjectLookupViewModel.cs

```diff
@@ -0,0 +1,67 @@
+using System.Diagnostics;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Windows.Input;
+using AngorApp.Model;
+using AngorApp.Services;
+using CSharpFunctionalExtensions;
+using ReactiveUI.SourceGenerators;
+using Serilog;
+using Zafiro.Avalonia.Controls.Navigation;
+using Zafiro.CSharpFunctionalExtensions;
+// o ReactiveUI.SourceGenerators, según tu setup
+
+namespace AngorApp.Sections.Browse.ProjectLookup;
+
+public partial class ProjectLookupViewModel : ReactiveObject, IProjectLookupViewModel
+{
+    [ObservableAsProperty] private ICommand? goToSelectedProject;
+
+    [ObservableAsProperty] private Maybe<IList<IProjectViewModel>> lookupResults;
+
+    [Reactive] private string? projectId;
+    [Reactive] private IProjectViewModel? selectedProject;
+
+    public ProjectLookupViewModel(
+        IProjectService projectService,
+        IWalletProvider walletProvider,
+        INavigator navigator,
+        UIServices uiServices)
+    {
+        Lookup = ReactiveCommand.CreateFromTask<string, Maybe<IList<IProjectViewModel>>>(
+            async pid =>
+            {
+                var maybeProject = await projectService.FindById(pid);
+                Log.Debug("Got project {ProjectId}", pid);
+
+                return maybeProject.Map<IProject, IList<IProjectViewModel>>(project =>
+                {
+                    var vm = new ProjectViewModel(walletProvider, project, navigator, uiServices);
+                    return new List<IProjectViewModel> { vm };
+                });
+            }
+        );
+
+        lookupResultsHelper = Lookup
+            .ObserveOn(RxApp.MainThreadScheduler)
```

Comment:
why do you need to use a schedular `MainThreadScheduler`? is it because there are calls over the wire?